server/routes/user.js

```diff
@@ -18,6 +18,7 @@ router.get("/profile", async (req, res, next) => {
 router.put("/profile", async (req, res, next) => {
   const { user_id: userId } = req.user;
   const { name, email, password } = req.body;
+
```

Comment:
Use your function checkUserByIDExist here, and depending on the result, return an error to the client. Remember to use await with checkUserByIDExist.